pkg/imagestore/imagestore.go

```diff
@@ -75,6 +78,12 @@ func NewImageStore(ed isoeditor.Editor, dataDir string, versions []map[string]st
 	if err := validateVersions(versions); err != nil {
 		return nil, err
 	}
+	// Create an HTTP Client that always skips TLS
+	transportConfig := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, //nolint:gosec // Solution is to ignore TLS (G402 error)
```

Comment:
Not sure if this was discussed elsewhere but, should this be configured? Not all deployments may want to run insecure TLS traffic. Am I missing something?

Reply:
Yeah, I suppose this should be configurable.
For example we probably don't ever want this behavior in the cloud.

Reply:
Good point! Will make this configurable

Reply:
What will be the default though? Currently we dont have a way to provide a certificate - so the user has not choice but running it in this mode. In the future, if we decide to add a certificate, it will make sense to have it as a configuration option, until then, u either have it insecure or serve it on http, so not sure how configuration will help here.

Reply:
@carbonin why do we care about the cloud - we deploy it, no? how would it point to different values then those that we provide...

Reply:
Right, and if the certs don't verify on `mirror.openshift.com` then I feel like something is wrong.
We still want MITM protection if we can get it.

People who need this feature are deploying the mirror themselves and know that we are not going to be able to verify that cert. We could make this (insecure) the default for the operator deploy or pass the configuration through the operator in some other way.

Another secure option would be to also have the file sha sum specified in os images. Then we can verify the content without being concerned that it isn't what we expected. These are provided on mirror.openshift.com (e.g. https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/latest/latest/sha256sum.txt), but this is a bit of a larger change.

Reply:
Is it best to verify certs by default?